### PR TITLE
Motion category page

### PR DESCRIPTION
**Pull request by mixth**
Response to #162 

## WIP
- [x] Dropdowns for filter and sort are not correctly styled ~~-> resources needed~~
- [x] Filter design for mobile -> no explicit design; interim solution
- [x] Badge to show motion count in category
- [x] Use higher resolution category icons ~~-> resources needed~~ -> improvement: using SVG instead of 100x100 px for all displays

#### Desktop
Screenshot_2020-06-18 Motions

#### iPad
Screenshot_2020-06-18 Motions(2)

#### Mobile
Screenshot_2020-06-18 Motions(1)

### CODE REVIEW

File: src/templates/motion-category-template.js [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    const motions = [...this.allMotions]
+    if (this.sortBy === "asc") {
+      motions.sort((a, b) => parseInt(a.id) > parseInt(b.id))
+    } else if (this.sortBy === "desc") {
+      motions.sort((a, b) => parseInt(a.id) < parseInt(b.id))
+    } else if (this.sortBy === "regis_no_desc") {
+      motions.sort(this.regisNoDesc)
+    } else if (this.sortBy === "regis_no_asc") {
+      motions.sort(this.regisNoAsc)
+    }
```
Comment (p16i) [inline src/templates/motion-category-template.js:61]: The branching is a bit lengthly. Maybe, we can create a sorter object containing all the sort type attributes.

For example,
```
const sorter  = {
   "asc": (a, b) => ... 
   "desc": (a, b) => ... 
   "regis_no_desc": (a, b) =>
}

motions.sort(sorter[this.sortBy])
```

File: src/templates/motion-category-template.js [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+      to_cabinet: "5. ส่งครม.",
+    }
+
+    this.setState({
+      motions: motions.filter(e => e.status === selectToStatus[this.filterBy]),
+    })
+  }
+
+  regisNoAsc = (a, b) => {
+    const dateA = this.convertDate(a.registration_no)
```
Comment (p16i) [inline src/templates/motion-category-template.js]: maybe we can convert the date once in the beginning?

**Comment by p16i** [comment]
This is a minor suggestion. Currently, it looks there is something wrong when there is no motion in the status that the user has selected: the page is simply empty. It would be good if we could show total motions of each status somehow. 

I'm thinking of having the number after each status. What do you think? 
image

File: src/templates/motion-category-template.js (hunk 1 above)
Comment (mixth) [reply src/templates/motion-category-template.js:61]: Definitely better. I was looking at it last night too, but didn't have time to consider it well enough. 😆

Thanks for all suggestions, will update soon.

Comment (p16i) [reply src/templates/motion-category-template.js:61]: imho, this is a minor improvement. we can also do it later.

**Comment by mixth** [comment]
> This is a minor suggestion. Currently, it looks there is something wrong when there is no motion in the status that the user has selected: the page is simply empty. It would be good if we could show total motions of each status somehow.
> 
> I'm thinking of having the number after each status. What do you think?
> image

Hmmmm... or at least showing some kind of message in the table instead of emptiness. I will bring this to the designer.

**Comment by rapee** [comment]
Agree on showing message telling there's no data for now krub.

Showing number requires a little more handling. 
Can you create a new issue? @heytitle

**Pull request merged**